plutus-core/plutus-core/src/PlutusCore/Parser.hs

```diff
@@ -46,7 +47,8 @@ lamTerm = withSpan $ \sp ->
 
 appTerm :: Parser PTerm
 appTerm = withSpan $ \sp ->
-    inBrackets $ mkIterApp sp <$> term <*> some term
+    -- TODO: should not use the same `sp` for all arguments.
```

Comment:
Not clear to me what it _should_ be. Do you want something like this?
```
[ f a b c ]
--> 
[[ f a ](0:5) b](0:7) c ](0:11)
```

Reply:
Yes, that's what I had in mind.